src/modules/preview-site/components/preview-site-row.tsx

```diff
@@ -84,7 +84,7 @@ export function PreviewSiteRow( { snapshot, selectedSite }: PreviewSiteRowProps
 				<div className="w-[51%]">
 					<div className="flex items-center">
 						<div className="text-[13px] leading-5 line-clamp-1 break-all">
-							{ selectedSite.name }
+							{ snapshot.name || selectedSite.name }
```

Comment:
We could follow the same pattern and display `Site Name Preview` as a fallback.

```suggestion
							{ /* translators: %s: Site name (e.g. "My Site Preview") */ }
							{ snapshot.name || sprintf( __( '%s Preview' ), selectedSite.name ) }
```

Reply:
Great idea! I like your proposa. 👍🏼